Assets/BossRoom/Scripts/Server/Game/State/ServerCharSelectState.cs

```diff
@@ -169,39 +170,55 @@ public override void NetworkStart()
             }
             else
             {
-                NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnected;
                 NetworkManager.Singleton.OnClientDisconnectCallback += OnClientDisconnectCallback;
                 CharSelectData.OnClientChangedSeat += OnClientChangedSeat;
 
-                if (IsHost)
-                {
-                    // host doesn't get an OnClientConnected()
-                    // and other clients could be connects from last game
-                    // So look for any existing connections to do intiial setup
-                    var clients = NetworkManager.Singleton.ConnectedClientsList;
-                    foreach (var net_cl in clients)
-                    {
-                        OnClientConnected(net_cl.ClientId);
-                    }
-                }
+                // Cosmin: Question? Should OnNotifyServerClientLoadedScene delegate also cover
+                // When a new client joins in? If yes, then this we will become obsolete for our case here
+                NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnected;
+                // NSS: Suggested Fix, use the OnNotifyServerClientLoadedScene to let you know that the client
```

Comment:
Do we need this in our code? We should try to cleanup these. Sure we can add doc on what's going on here, but right now new users might be a bit confused by these?